File: backend/src/main/java/com/rainbowdev/backend/service/TaskService.java
package com.rainbowdev.backend.service;

import com.rainbowdev.backend.model.Task;
import com.rainbowdev.backend.repository.TaskRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class TaskService {

    @Autowired
    private TaskRepository taskRepository;

    public List<Task> getAllTasks() {
        return taskRepository.findAll();
    }

    public Task getTaskById(Long id) {
        return taskRepository.findById(id).orElseThrow(() -> new RuntimeException("Task not found"));
    }

    public Task createTask(Task task) {
        System.out.println("Versuche Task zu erstellen: " + task);
        try {
            Task savedTask = taskRepository.save(task);
            System.out.println("Task erfolgreich erstellt: " + savedTask);
            return savedTask;
        } catch (Exception e) {
            System.err.println("Fehler beim Erstellen des Tasks: " + e.getMessage());
            e.printStackTrace();
            throw e;
        }
    }

    public Task updateTask(Long id, Task taskDetails) {
        Task task = taskRepository.findById(id).orElseThrow(() -> new RuntimeException("Task not found"));
        task.setTitle(taskDetails.getTitle());
        task.setDescription(taskDetails.getDescription());
        task.setDueDate(taskDetails.getDueDate());
        task.setCompleted(taskDetails.isCompleted());
        return taskRepository.save(task);
    }

    public boolean deleteTask(Long id) {
        Task task = taskRepository.findById(id).orElseThrow(() -> new RuntimeException("Task not found"));
        taskRepository.delete(task);
        return false;
    }
}

File: frontend/src/services/taskService.ts
import axios from 'axios';
import { API_BASE_URL } from '../config/environment';
import { Task } from '../types/task';

const taskService = {
    getAllTasks: async (): Promise<Task[]> => {
        const response = await axios.get(`${API_BASE_URL}/tasks`);
        return response.data;
    },

    getTaskById: async (id: number): Promise<Task> => {
        const response = await axios.get(`${API_BASE_URL}/tasks/${id}`);
        return response.data;
    },

    createTask: async (task: Omit<Task, 'id'>): Promise<Task> => {
        const response = await axios.post(`${API_BASE_URL}/tasks`, task);
        return response.data;
    },

    updateTask: async (id: number, task: Task): Promise<Task> => {
        const response = await axios.put(`${API_BASE_URL}/tasks/${id}`, task);
        return response.data;
    },

    deleteTask: async (id: number): Promise<void> => {
        await axios.delete(`${API_BASE_URL}/tasks/${id}`);
    },
};

export default taskService;